diffsims/crystallography/_diffracting_vector.py

```diff
@@ -0,0 +1,153 @@
+        required.
+    hkl : numpy.ndarray, list, or tuple, optional
+        Indices of reciprocal lattice vector(s). Default is ``None``.
+        This, ``xyz``, or ``hkil`` is required.
+    hkil : numpy.ndarray, list, or tuple, optional
+        Indices of reciprocal lattice vector(s), often preferred over
+        ``hkl`` in trigonal and hexagonal lattices. Default is ``None``.
+        This, ``xyz``, or ``hkl`` is required.
+    intensity : numpy.ndarray, list, or tuple, optional
+        Intensity of the diffraction vector(s). Default is ``None``.
+    rotation : orix.quaternion.Rotation, optional
+        Rotation matrix previously applied to the reciprocal lattice vector(s) and the
+        lattice of the phase. Default is ``None`` which corresponds to the
+        identity matrix.
+
+
+    Examples
+    --------
+    >>> from diffpy.structure import Atom, Lattice, Structure
+    >>> from orix.crystal_map import Phase
+    >>> from diffsims.crystallography import DiffractingVector
+    >>> phase = Phase(
+    ...     "al",
+    ...     space_group=225,
+    ...     structure=Structure(
+    ...         lattice=Lattice(4.04, 4.04, 4.04, 90, 90, 90),
+    ...         atoms=[Atom("Al", [0, 0, 1])],
+    ...     ),
+    ... )
+    >>> rlv = DiffractingVector(phase, hkl=[[1, 1, 1], [2, 0, 0]])
+    >>> rlv
+    ReciprocalLatticeVector (2,), al (m-3m)
+    [[1. 1. 1.]
+     [2. 0. 0.]]
+
+    """
+
+    def __init__(
+        self, phase, xyz=None, hkl=None, hkil=None, intensity=None, rotation=None
+    ):
+        super().__init__(phase, xyz=xyz, hkl=hkl, hkil=hkil)
+        if intensity is None:
+            self._intensity = np.full(self.shape, np.nan)
+        elif len(intensity) != self.size:
+            raise ValueError("Length of intensity array must match number of vectors")
+        else:
+            self._intensity = np.array(intensity)
+        self._rotation = rotation
+
+    def __getitem__(self, key):
+        dv_new = super().__getitem__(key)
+        if np.isnan(self.intensity).all():
+            dv_new._intensity = np.full(dv_new.shape, np.nan)
+        else:
+            slic = self.intensity[key]
+            if not hasattr(slic, "__len__"):
+                slic = np.array(
+                    [
+                        slic,
+                    ]
+                )
+            dv_new._intensity = slic
+
+        return dv_new
+
+    @property
+    def intensity(self):
+        return self._intensity
+
+    @intensity.setter
+    def intensity(self, value):
+        if not hasattr(value, "__len__"):
+            value = np.array(
+                [
+                    value,
+                ]
+                * self.size
+            )
+        if len(value) != self.size:
+            raise ValueError("Length of intensity array must match number of vectors")
+        self._intensity = np.array(value)
+
+    @property
+    def lattice_aligned_data(self):
```

Comment:
So these vectors are the same as they were before rotated by a rotation. Can't we *separately* keep track of a set of starting vectors, a rotation, and a set of rotated vectors? Why do these separate things have to be stored in a single class instance?

Reply:
Hmm.... I think this is related to above.  What use is the RLV if it __doesn't__ have information about the Rotation otherwise it is just the same as a `Vector3D` with confusing hkl values that don't really make physical sense and look correct because they are rounded.